HeartbeatLogging/Tests/Unit/HeartbeatControllerTests.swift

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import HeartbeatLogging
+
+class HeartbeatControllerTests: XCTestCase {
+  func testFlushWhenEmpty() throws {
+    // Given
+    let controller = HeartbeatController(storage: HeartbeatStorageFake())
+    // When
+    let flushed = controller.flush()
+    // Then
+    XCTAssertNil(flushed)
+  }
+
+  func testLogThenFlush() throws {
+    // Given
+    let controller = HeartbeatController(storage: HeartbeatStorageFake())
+    XCTAssertNil(controller.flush())
+    // When
+    controller.log(#function)
+    // Then
+    XCTAssertNotNil(controller.flush())
```

Comment:
TODO: Validate the heartbeat content to match data returned by `HeartbeatStorageFake`

Reply:
I have some broader questions about what this should look like. I'm going to place a TODO at the top of this file and come back to it in a follow-up PR. 

I'm working on the next set of unit tests (for `HeartbeatInfo` data model) and my learnings there may inform how this is done.